simtools/corsika/corsika_default_config.py

```diff
@@ -0,0 +1,217 @@
+from collections import defaultdict
+
+import astropy.units as u
+from scipy.interpolate import interp1d
+
+
+class CorsikaDefaultConfig:
+    """
+    This class contains the default configuration for CORSIKA parameters for
+    the various primary particles. It includes all basic dependencies on zenith angles, etc.
+    """
+
+    def __init__(self, primary=None, zenith_angle=None):
+        """
+        Initialize CorsikaDefaultConfig.
+        """
+        self._primary = primary
+        self._zenith_angle = zenith_angle
+        self._energy_slope = -2.0
+        self.energy_ranges = self._define_hardcoded_energy_ranges()
+        self.number_of_showers = self._define_hardcoded_number_of_showers()
+
+    @property
+    def primary(self):
+        """
+        Primary particle.
+        """
+        return self._primary
+
+    @primary.setter
+    def primary(self, primary):
+        """
+        Set primary particle.
+        """
+        supported_primaries = [
+            "gamma",
+            "gamma_diffuse",
+            "electron",
+            "proton",
+            "muon",
+            "helium",
+            "nitrogen",
+            "silicon",
+            "iron",
+        ]
+        if primary not in supported_primaries:
+            raise ValueError(
+                f"Invalid primary particle: {primary}. Must be one of {supported_primaries}"
+            )
+        self._primary = primary
+
+    @property
+    def zenith_angle(self):
+        """
+        Zenith angle.
+        """
+        return self._zenith_angle
+
+    @zenith_angle.setter
+    @u.quantity_input(zenith_angle=u.deg)
+    def zenith_angle(self, zenith_angle):
+        """
+        Set zenith angle.
+        """
+        allowed_zenith_angle_interval = [20.0, 60.0] * u.deg
+        if (
+            zenith_angle < allowed_zenith_angle_interval[0]
+            or zenith_angle > allowed_zenith_angle_interval[1]
+        ):
+            raise ValueError(
+                f"The zenith angle, {zenith_angle:.1f}, "
+                f"is outside of the allowed interval, {allowed_zenith_angle_interval}."
+                "This interval is enforced because values can only be interpolated "
+                "between zenith angles that were manually optimized."
+            )
+
+        self._zenith_angle = zenith_angle
+
+    @property
+    def energy_slope(self):
+        """
+        Energy slope.
+        """
+        return self._energy_slope
+
+    def _define_hardcoded_energy_ranges(self):
+        """
+        Define the hardcoded energy ranges for the various primaries.
```

Comment:
Do you want to add in the docstring (or in the module docstring) that these default values have been derived to cover the full CTAO energy range and assume that all three telescopes types are simulated?

Reply:
Yes, I will add it to the docstring in both places. BTW, the energy range does not change if SSTs are simulated or not (only the area).